osmaxx-py/osmaxx/excerptexport/tests/test_conversion_api_client.py

```diff
@@ -79,27 +80,31 @@ def test_create_job(self):
         self.assertEqual(self.extraction_order.process_id, response.json().get('rq_job_id'))
         self.assertIsNotNone(self.extraction_order.process_id)
 
-    @vcr.use_cassette('fixtures/vcr/conversion_api-test_download_files.yml')
     def test_download_files(self):
-        api_client = ConversionApiClient()
-        api_client.create_job(self.extraction_order)
-        # HACK: enable this line if testing against a new version of the api, otherwise vcr records the wrong answer!
-        # sleep(120)
-        success = api_client.download_result_files(self.extraction_order)
-        self.assertIsNone(api_client.errors)
-
-        self.assertTrue(success)
-        self.assertEqual(self.extraction_order.output_files.count(), 2)
-        self.assertEqual(self.extraction_order.output_files.order_by('id')[0].content_type, 'fgdb')
-        self.assertEqual(self.extraction_order.output_files.order_by('id')[1].content_type, 'spatialite')
-        self.assertEqual(
-            len(self.extraction_order.output_files.order_by('id')[0].file.read()),
-            446013
-        )
-        self.assertEqual(
-            len(self.extraction_order.output_files.order_by('id')[1].file.read()),
-            368378
-        )
+        with vcr.use_cassette('fixtures/vcr/conversion_api-test_download_files.yml') as cassette:
+            api_client = ConversionApiClient()
+            api_client.create_job(self.extraction_order)
+
+            if cassette.dirty:
+                # wait for external service to complete request
+                time.sleep(120)
```

Comment:
Looking at [the code of `Cassette`](https://github.com/kevin1024/vcrpy/blob/1324feae9920e449e44a987ad178a8ec799a413f/vcr/cassette.py#L148), our use of `dirty` (indicating whether the last response came from the record or has been freshly recorded just now) doesn't match that attribute's intended purpose (indicating whether the cassette object contains unsaved request-response pairs, i.e., not yet persisted to the cassette file).

If we want to keep (ab)using `dirty`, we should have a separate test in place that ensures it keeps working as we want.